lib/bundler/cli/add.rb

```diff
@@ -11,7 +11,14 @@ def initialize(options, gem_name)
 
     def run
       version = @options[:version].nil? ? nil : @options[:version].split(",").map(&:strip)
-      dependency = Bundler::Dependency.new(@gem_name, version, @options)
+
+      begin
+        dependency = Bundler::Dependency.new(@gem_name, version, @options)
+      rescue Gem::Requirement::BadRequirementError => e
+        Bundler.ui.error(e.message)
+        return
```

Comment:
Shouldn't the messages passed by `Gem::Requirement::BadRequirementError` be descriptive enough? It would output something like: `Illformed requirement ["~> 1 . 0"]`